Remove outdated feature tours and on demand tours.

benbowler opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #10574

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

10upsimon reviewed (approved):
@benbowler changes LGTM ✅  Moving to MR

aaemnnosttv commented on assets/js/components/KeyMetrics/hooks/useChangeMetricsFeatureTourEffect.js:
```diff
@@ -1,59 +0,0 @@
-
-	useEffect( () => {
-		if ( renderChangeMetricLink && isUserEligibleForTour ) {
-			triggerOnDemandTour( sharedKeyMetrics );
```
Looking at this again, I think we may want to keep this one (or repurpose into a tooltip maybe) since it isn't really a standard tour that is only shown to users who upgraded from an older version of SK.

aaemnnosttv replied:
Confirmed that we do want to keep this one. I'll update the AC

benbowler replied:
Sounds good, I raised this in the IB originally as I agree some tours that appear when a new feature is enabled make sense to keep longer term.

aaemnnosttv reviewed (commented):
Pending question to Mariya

aaemnnosttv reviewed (changes requested):
Thanks @benbowler, and apologies for the change but we actually want to keep the on-demand shared key metrics tour. All the rest can go though. I'll update the AC to reflect this change.

benbowler commented:
@aaemnnosttv I reverted this tour, while testing I found that the class has been removed from the ChangeMetricsLink which means this tour has been broken for a little while. I fixed this as part of the update and added an inline comment to make sure it's not removed in future.

eugene-manuilov reviewed (changes requested):
LGMT. Just one change is suggested to remove an excessive comment. I'll apply it myself.
